src/charm.py

```diff
@@ -571,6 +542,100 @@ def _gnb_subnet_config_is_valid(self) -> bool:
     def _get_gnb_subnet_config(self) -> Optional[str]:
         return self.model.config.get("gnb-subnet")
 
+    def _get_access_nad_config(self) -> Dict[Any, Any]:
+        return {
+            "cniVersion": "0.3.1",
+            "ipam": {
+                "type": "static",
+                "routes": [
+                    {
+                        "dst": self._get_gnb_subnet_config(),
+                        "gw": self._get_access_network_gateway_ip_config(),
+                    },
+                ],
+                "addresses": [
+                    {
+                        "address": self._get_access_network_ip_config(),
+                    }
+                ],
+            },
+            "capabilities": {"mac": True},
+        }
+
+    def _get_core_nad_config(self) -> Dict[Any, Any]:
+        return {
+            "cniVersion": "0.3.1",
+            "ipam": {
+                "type": "static",
+                "addresses": [
+                    {
+                        "address": self._get_core_network_ip_config(),
+                    }
+                ],
+            },
+            "capabilities": {"mac": True},
+        }
+
+    def _get_core_interface_mtu_config(self) -> Optional[str]:
+        return self.model.config.get("core-interface-mtu-size")
+
+    def _get_access_interface_mtu_config(self) -> Optional[str]:
+        return self.model.config.get("access-interface-mtu-size")
+
+    def _access_interface_mtu_size_is_valid(self) -> bool:
+        """Checks whether the access interface MTU size is valid.
+
+        Returns:
+            bool: Whether access interface MTU size is valid
+        """
+        if access_interface_mtu_size := self._get_access_interface_mtu_config() is None:
+            return True
+        return self._mtu_size_is_valid(int(access_interface_mtu_size))
+
+    def _core_interface_mtu_size_is_valid(self) -> bool:
+        """Checks whether the core interface MTU size is valid.
+
+        Returns:
+            bool: Whether core interface MTU size is valid
+        """
+        if core_interface_mtu_size := self._get_core_interface_mtu_config() is None:
+            return True
+        return self._mtu_size_is_valid(int(core_interface_mtu_size))
+
+    @staticmethod
+    def _mtu_size_is_valid(mtu_size: Any) -> bool:
```

Comment:
How about enforcing MTU being `int` on the signature level?

Reply:
This static method is removed.